pkg/build/builder/util.go

```diff
@@ -26,3 +30,67 @@ func getBuildEnvVars(build *api.Build) map[string]string {
 	}
 	return envVars
 }
+
+type BuildConfigClientInterface interface {
+	GetBuildConfig(namespace, name string) (*buildapi.BuildConfig, error)
+	UpdateBuildConfig(buildConfig *buildapi.BuildConfig) error
+}
+
+type BuildClientInterface interface {
```

Comment:
This interface purpose is to provide external services to the controller. I'd propose moving `CreateBuildFromConfig` and `CreateBuildWithImmutableId` methods to the controller, because they contain logic that should be part of the controller. The only interaction with other parts of the system I see there is `CreateBuild` which is part of that interface and all the rest should happen inside controller IMHO.